### PR TITLE
Fix #791 : Lowfi Landscape: OngoingTopicListActivity

### LINKED ISSUE #791: Lowfi Landscape: OngoingTopicListActivity
Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/ce21cbf2-0439-47d7-bb37-005cfc6b9742/UP-Topics-in-Progress-

Also, please add test case for checking that item is visible in orientation change also. You will mainly need to do this in `OngoingTopicListActivityTest`

### PR DESCRIPTION
**Pull request by virajpatel2411**
## Explanation
Fixes #791 
Landscape mode for ongoing topics list activity created according to mocks.
Test Case for checking the working by changing orientation added.

## Mock Link : 
https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/ce21cbf2-0439-47d7-bb37-005cfc6b9742/UP-Topics-in-Progress-

### CODE REVIEW

**Comment by virajpatel2411** [comment]
@rt4914 @veena14cs @nikitamarysolomanpvt  Please review it. I have made the requested changes.

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt [hunk 1]
```diff
@@ -132,14 +136,27 @@ class OngoingTopicListActivityTest {
     }
   }
 
+  @Test
+  fun testTopicPracticeFragment_loadFragment_changeOrientation_textIsCorrect() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: This test case is failing.

Please take reference from this test case: `testOngoingTopicList_checkItem1_titleIsCorrect`

Also, in your previous PR which you have closed, I had mentioned some changes also, please do those changes here too.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt (hunk 1 above)
Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: Does I have to do the same thing as in  `testOngoingTopicList_checkItem1_titleIsCorrect` . Like get to the position of recycler view and then check if it contains substring or not?
@rt4914

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: @virajpatel2411
Yes but with one step added for orientation change.
`testOngoingTopicList_changeConfiguration_checkItem1_titleIsCorrect`

Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: Okayy

Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: @rt4914 
Is this okay?
`  @Test
  fun testTopicPracticeFragment_loadFragment_changeOrientation_textIsCorrect() {
    launch<OngoingTopicListActivity>(createOngoingTopicListActivityIntent(internalProfileId)).use {
      onView(withId(R.id.ongoing_topic_list)).perform(
        scrollToPosition<RecyclerView.ViewHolder>(
          1
        )
      )
      onView(atPositionOnView(R.id.ongoing_topic_list, 1, R.id.topic_name_text_view)).check(
        matches(
          withText(containsString("Ratios"))
        )
      )
    }
  }`

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: @virajpatel2411 Actually it is not correct to check like this, make a commit and then I will check this in android studio.

Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: Okayy

**Comment by virajpatel2411** [comment]
@rt4914 @veena14cs @nikitamarysolomanpvt Can you please review the test case?

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt (hunk 1 above)
Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:138]: I have made a commit. Please check @rt4914.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Test cases are failing and also, the orientation for this activity is still locked to `portrait` in `AndroidManifest` file.
To test the code that you working on your device whether the orientation is change or not, you can follow these steps:
Reference: https://gist.github.com/rt4914/5258375c8d2cca81b6a5b62ccab8e143
you can use `StoryProgressTestHelper` to mark dummy chapters as completed so that you can see items in `OngoingTopicListActivity`. 
You will need following lines: 39, 92, 93 from the above link in your `HomeFragmentPresenter`.

Also, please do not code without understanding the requirement that you are working on.

**Comment by virajpatel2411** [comment]
> Test cases are failing and also, the orientation for this activity is still locked to `portrait` in `AndroidManifest` file.
> To test the code that you working on your device whether the orientation is change or not, you can follow these steps:
> Reference: https://gist.github.com/rt4914/5258375c8d2cca81b6a5b62ccab8e143
> you can use `StoryProgressTestHelper` to mark dummy chapters as completed so that you can see items in `OngoingTopicListActivity`.
> You will need following lines: 39, 92, 93 from the above link in your `HomeFragmentPresenter`.
> 
> Also, please do not code without understanding the requirement that you are working on.

So, the reason for the test case failing is the locked potrait mode only or some other thing also?@rt4914

**Comment by rt4914** [comment]
> > Test cases are failing and also, the orientation for this activity is still locked to `portrait` in `AndroidManifest` file.
> > To test the code that you working on your device whether the orientation is change or not, you can follow these steps:
> > Reference: https://gist.github.com/rt4914/5258375c8d2cca81b6a5b62ccab8e143
> > you can use `StoryProgressTestHelper` to mark dummy chapters as completed so that you can see items in `OngoingTopicListActivity`.
> > You will need following lines: 39, 92, 93 from the above link in your `HomeFragmentPresenter`.
> > Also, please do not code without understanding the requirement that you are working on.
> 
> So, the reason for the test case failing is the locked potrait mode only or some other thing also?@rt4914

Not actually, because test cases are failing because of incorrect test -case code. But my concern here is not about test case but about making sure that you can actually see `OngoingTopicListActivity` with content when you run your app.

**Comment by virajpatel2411** [comment]
> > > Test cases are failing and also, the orientation for this activity is still locked to `portrait` in `AndroidManifest` file.
> > > To test the code that you working on your device whether the orientation is change or not, you can follow these steps:
> > > Reference: https://gist.github.com/rt4914/5258375c8d2cca81b6a5b62ccab8e143
> > > you can use `StoryProgressTestHelper` to mark dummy chapters as completed so that you can see items in `OngoingTopicListActivity`.
> > > You will need following lines: 39, 92, 93 from the above link in your `HomeFragmentPresenter`.
> > > Also, please do not code without understanding the requirement that you are working on.
> > 
> > 
> > So, the reason for the test case failing is the locked potrait mode only or some other thing also?@rt4914
> 
> Not actually, because test cases are failing because of incorrect test -case code. But my concern here is not about test case but about making sure that you can actually see `OngoingTopicListActivity` with content when you run your app.

I am getting the error shown in the below screenshot. The tests are failing due to the setUp() method.
Screenshot (55)

**Comment by virajpatel2411** [comment]
> > > Test cases are failing and also, the orientation for this activity is still locked to `portrait` in `AndroidManifest` file.
> > > To test the code that you working on your device whether the orientation is change or not, you can follow these steps:
> > > Reference: https://gist.github.com/rt4914/5258375c8d2cca81b6a5b62ccab8e143
> > > you can use `StoryProgressTestHelper` to mark dummy chapters as completed so that you can see items in `OngoingTopicListActivity`.
> > > You will need following lines: 39, 92, 93 from the above link in your `HomeFragmentPresenter`.
> > > Also, please do not code without understanding the requirement that you are working on.
> > 
> > 
> > So, the reason for the test case failing is the locked potrait mode only or some other thing also?@rt4914
> 
> Not actually, because test cases are failing because of incorrect test -case code. But my concern here is not about test case but about making sure that you can actually see `OngoingTopicListActivity` with content when you run your app.

Yes I am being able to see the content in landscape mode on running the app. Screenshot is

**Comment by virajpatel2411** [comment]
> Test cases are failing and also, the orientation for this activity is still locked to `portrait` in `AndroidManifest` file.
> To test the code that you working on your device whether the orientation is change or not, you can follow these steps:
> Reference: https://gist.github.com/rt4914/5258375c8d2cca81b6a5b62ccab8e143
> you can use `StoryProgressTestHelper` to mark dummy chapters as completed so that you can see items in `OngoingTopicListActivity`.
> You will need following lines: 39, 92, 93 from the above link in your `HomeFragmentPresenter`.
> 
> Also, please do not code without understanding the requirement that you are working on.

I have taken reference from the other test case and corrected mine. I am commiting the test case please have a look.@rt4914

File: app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt [hunk 2]
```diff
@@ -66,6 +68,9 @@ class HomeFragmentPresenter @Inject constructor(
     itemList.add(allTopicsViewModel)
     topicListAdapter = TopicListAdapter(activity, itemList, promotedStoryList)
 
+    storyProgressTestHelper.markFullStoryProgressForFractions(profileId)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt]: As mentioned earlier, this is just for testing purpose so that you can actually see the content we do not this in the app. You will need to revert these changes in this file.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -296,4 +296,5 @@
   <!-- WalkthroughActivity -->
   <string name="walkthrough_welcome_description">Learn new math skills with stories that show you how to use them in your daily life</string>
   <string name="welcome">"Welcome %s!"</string>
+  <string name="topic_name_text_view">Ratios and Proportional Reasoning</string>
```
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: Remove this string as it is not getting used in production code. It is just getting used in test code and it is completely fine if we hard code this in test code. This is how it works:
If you are writing a test case in which you have a string to compare:
1. Check if the string already exists in `strings.xml`, if yes then use that resource in test code.
2. If not, then hard-code it in test case but do not introduce it in `strings.xml` file.

`strings.xml` will contain only those strings which are getting used in production code.

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt [hunk 4]
```diff
@@ -132,14 +135,42 @@ class OngoingTopicListActivityTest {
...
+        scrollToPosition<RecyclerView.ViewHolder>(
+          0
+        )
+      )
+      onView(atPositionOnView(R.id.ongoing_topic_list, 1, R.id.topic_name_text_view)).check(
+        matches(
+          withText(containsString("Ratios and Proportional Reasoning"))
+        )
+      )
+      onView(isRoot()).perform(OrientationChangeAction.orientationLandscape())
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: For better code readability import functions directly:
`onView(isRoot()).perform(orientationLandscape())`

Note. once you make this change make sure that you remove all unnecessary imports.

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt [hunk 5]
```diff
@@ -132,14 +135,42 @@ class OngoingTopicListActivityTest {
...
 
+  @Test
+  fun testTopicPracticeFragment_loadFragment_changeOrientation_textIsCorrect() {
+    launch<OngoingTopicListActivity>(createOngoingTopicListActivityIntent(internalProfileId)).use {
+      onView(withId(R.id.ongoing_topic_list)).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          0
+        )
+      )
+      onView(atPositionOnView(R.id.ongoing_topic_list, 1, R.id.topic_name_text_view)).check(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: Test case is failing because you index is incorrect. Instead of `1` it should be 0.

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt [hunk 6]
```diff
@@ -132,14 +135,42 @@ class OngoingTopicListActivityTest {
...
+          withText(containsString("Ratios and Proportional Reasoning"))
+        )
+      )
+      onView(isRoot()).perform(OrientationChangeAction.orientationLandscape())
+      onView(withId(R.id.ongoing_topic_list)).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          0
+        )
+      )
+      onView(atPositionOnView(R.id.ongoing_topic_list, 1, R.id.topic_name_text_view)).check(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: Test case is failing because you index is incorrect. Instead of `1` it should be 0.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Changes suggested. Your test case is failing. Please make sure that your test case pass before assigning it for review.

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt (hunk 4 above)
Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: Done

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (virajpatel2411) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt (hunk 2 above)
Comment (virajpatel2411) [reply app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt]: Removed

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt (hunk 5 above)
Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: Changed

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt (hunk 6 above)
Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: Changed

**Comment by virajpatel2411** [comment]
@rt4914 I have made the changes. Can you please check the test case?

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt [hunk 7]
```diff
@@ -132,14 +135,42 @@ class OngoingTopicListActivityTest {
...
+        matches(
+          withText(containsString("Ratios and Proportional Reasoning"))
+        )
+      )
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.ongoing_topic_list)).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          0
+        )
+      )̥
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: If you notice this, this is not a bracket, this is some special character which is not in English. There must have been some typo while writing code. Please remove this.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL.

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt (hunk 7 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: There is a `circle/dot` below bracket

Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: Removed.. Can you check now @rt4914

**Review by rt4914 — APPROVED** [review]
LGTM, but this PR is blocked as CircleCI has run out of memory mostly. As the tests are failing unexpectedly.
We can solve this by making a simple commit to this PR. But I let's do this at a later stage when there is less load on circle CI. I will inform you when you when to make a commit.

**Comment by virajpatel2411** [comment]
Okayy @rt4914

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt [hunk 8]
```diff
@@ -44,7 +46,8 @@ import javax.inject.Singleton
 class OngoingTopicListActivityTest {
 
   private val internalProfileId = 0
-  @Inject lateinit var storyProfileTestHelper: StoryProgressTestHelper
+  @Inject
+  lateinit var storyProfileTestHelper: StoryProgressTestHelper
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: @virajpatel2411 As discussed earlier, we were facing some issues with CircleCI, so for you you can make this change and push the code to this branch.

This should be in single line, something line this:
`@Inject lateinit var storyProfileTestHelper: StoryProgressTestHelper`

**Review by rt4914 — APPROVED** [review]
@virajpatel2411 PTAL

File: app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt (hunk 8 above)
Comment (virajpatel2411) [reply app/src/sharedTest/java/org/oppia/app/ongoingtopiclist/OngoingTopicListActivityTest.kt]: @rt4914 Done the required change and committed.

**Comment by rt4914** [comment]
Closing this PR as it has been recreated at #860
